app/controllers/application_controller.rb

```diff
@@ -140,13 +140,19 @@ def tos_skippable?
 
   def post_or_reply_link(reply)
     return unless reply.id.present?
-    if reply.is_a?(Reply)
-      reply_path(reply, anchor: "reply-#{reply.id}")
+    post_or_reply_mem_link({id: reply.id, class: reply.class})
+  end
+  helper_method :post_or_reply_link
+
+  def post_or_reply_mem_link(obj)
```

Comment:
Tiny nit: we could use keyword parameters with defaults of `nil` instead of a hash, maybe? That way it's clearer that we're only getting `id` and `class`, if anything.

Reply:
Might make the code a bit uglier in the partial, though, so it's up to you!